pkg/kubelet/metrics/collectors/volume_stats.go

```diff
@@ -61,6 +61,12 @@ var (
 		[]string{"namespace", "persistentvolumeclaim"}, nil,
 		metrics.ALPHA, "",
 	)
+
+	volumeStatsHealthAbnormalDesc = metrics.NewDesc(
+		metrics.BuildFQName("", kubeletmetrics.KubeletSubsystem, kubeletmetrics.VolumeStatsHealthStatusKey),
+		"Volume health status. The count is either 1 or 0",
+		[]string{"namespace", "persistentvolumeclaim"}, nil,
```

Comment:
This was the issue that @logicalhan mentioned https://github.com/kubernetes/kubernetes/issues/104008. But it was quite different since it was adding a namespace dimension to a request-related metric which is super expensive.

@xing-yang what do you think about having the abnormal status exposed in the PVC object directly and getting the metric via[ kube-state-metrics](https://github.com/kubernetes/kube-state-metrics)? It seems more appropriate to me.

Reply:
Thanks @dgrisonnet. We discussed about adding abnormal status as a first class field in PVC status and use that for reaction. We could not reach agreement on that because we could not reach consensus on how to use that field for reaction. So there will be continued discussions on that and that's tracked as a separate task now. 
When discussing the KEP for this PR, we have agreed to expose abnormal status as metric as implemented here.
@gnufied do you want to share your use case for this?

Reply:
One of goals I had was - having a uniform metric for PV health across k8s ecosystem. Coming from kubelet will ensure that this happens. Having it in different optional component means it can't be standardised.  

For example - current pvc usage metrics (which also has name and namespace as metric label) has facilitated an ecosystem of monitoring tools and operators which can act on a uniform metric - even though core k8s is not yet sure how to represent "used/availale" space in core PVC objects. So basically I can pick an operator that can automatically expand a PVC if available space runs low or alert on it. It does not matter which volume type it is but since core k8s has standardized on usage metrics, we all benefit from it.

My use case is similar for health metrics. Sure - other components can emit it and it can take awhile to formalize API change necessary to represent PV health in PVC's status but having a metric gets us halfway there.

Reply:
@fengzixu Can you address @gnufied's comment here and add `volume_health_status` as a label?

Reply:
Sounds good from my side :+1:

@xing-yang @fengzixu as discussed previously in https://github.com/kubernetes/kubernetes/pull/105585#discussion_r737634001, `condition` might a better suited here.

Reply:
I'm not sure if "condition" is better than volume health.  We have always named this feature "volume health monitoring" and the alpha feature gate is called "CSIVolumeHealth" which was introduced in v1.21 release: https://kubernetes.io/docs/concepts/storage/volume-health-monitoring/.  From the controller side, the name of the controller is "External Health Monitor controller" and the repo is https://github.com/kubernetes-csi/external-health-monitor.

Reply:
I preferred `condition` because:
1. It is shorter. :-)
2. Because we use `volume_condition` string in CSI spec and hence is similarly named
3. Because we use `condition` string all over the place in k8s to report status of things such as operators, pods etc. It seems natural to use same string for volume health. 

But I could live with `health_status` too.

Reply:
Hey guys. I will choose `volume_health_status` as the label of this metrics